### PR TITLE
Specialized exception for non-unique values

### PR DESCRIPTION
**Pull request by flexatone**
Implements request of #7

### CODE REVIEW

File: test_automap.py [hunk 1]
```diff
@@ -78,3 +78,11 @@ def test_issue_3(keys: Keys) -> None:
...
     with pytest.raises(ValueError):
         a |= (key,)
+
+
+def test_non_unique_exception():
+    with pytest.raises(ValueError):
+        _ = automap.AutoMap(("a", "b", "a"))
+
+    with pytest.raises(automap.NonUniqueError):
+        _ = automap.AutoMap(("a", "b", "a"))
```
Comment (brandtbucher) [inline test_automap.py:88]: I'd like to hypothesize this, if possible!

We can follow the pattern in the existing tests. Something like this:

```suggestion
@hypothesis.given(keys=hypothesis.infer)
def test_non_unique_exception(keys: Keys) -> None:
    hypothesis.assume(keys)
    duplicate = next(iter(keys))
    with pytest.raises(ValueError):
        automap.AutoMap([*keys, duplicate])

    with pytest.raises(automap.NonUniqueError):
        automap.AutoMap([*keys, duplicate])
```

File: automap.c [hunk 2]
```diff
@@ -510,7 +503,7 @@ insert(FAMObject *self, PyObject *key, Py_ssize_t offset, Py_hash_t hash)
         return -1;
     }
     if (self->table[index].hash != -1) {
-        PyErr_SetObject(PyExc_ValueError, key);
+        PyErr_Format(NonUniqueError, "Non-unique value found: %R", key, NULL);
```
Comment (brandtbucher) [inline automap.c:506]:
```suggestion
        PyErr_Format(NonUniqueError, "Non-unique value found: %R", key);
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
Thanks! Just a couple minor suggestions. If they look good but you're too busy, just let me know and I'll take care of them:

File: test_automap.py (hunk 1 above)
Comment (flexatone) [reply test_automap.py:88]: Agreed, this is much better.

**Comment by flexatone** [comment]
Many thanks for the feedback; I have made these changes.

**Comment by brandtbucher** [comment]
Thanks!

**Pull request merged**